compiler/src/main/java/com/squareup/anvil/compiler/codegen/dagger/AssistedFactoryGenerator.kt

```diff
@@ -0,0 +1,232 @@
+    val className = "${clazz.generateClassName()}_Impl"
+    val implClass = ClassName(packageName, className)
+
+    val typeParameters = clazz.typeParameterList
+      ?.parameters
+      ?.mapNotNull { parameter ->
+        val extendsBound = parameter.extendsBound?.fqNameOrNull(module)?.asClassName(module)
+        if (parameter.fqNameOrNull(module) == null) {
+          TypeVariableName(parameter.nameAsSafeName.asString(), listOfNotNull(extendsBound))
+        } else {
+          null
+        }
+      }
+      ?: emptyList()
+
+    fun ClassName.parameterized(): TypeName {
+      return if (typeParameters.isEmpty()) this else parameterizedBy(typeParameters)
+    }
+
+    val generatedFactory = FqName(returnType.fqNameSafe.asString() + "_Factory")
+      .asClassName(module).parameterized()
+
+    val classType = clazz.asClassName().parameterized()
+    val delegateFactoryName = "delegateFactory"
+
+    val content = FileSpec.buildFile(packageName, className) {
+      TypeSpec.classBuilder(implClass)
+        .apply {
+          typeParameters.forEach { addTypeVariable(it) }
+
+          if (DescriptorUtils.isInterface(classDescriptor)) {
+            addSuperinterface(classType)
+          } else {
+            superclass(classType)
+          }
+
+          primaryConstructor(
+            FunSpec.constructorBuilder()
+              .addParameter(delegateFactoryName, generatedFactory)
+              .build()
+          )
+
+          addProperty(
+            PropertySpec.builder(delegateFactoryName, generatedFactory)
+              .initializer(delegateFactoryName)
+              .addModifiers(PRIVATE)
+              .build()
+          )
+        }
+        .addFunction(
+          FunSpec.builder("create")
+            .addModifiers(OVERRIDE)
+            .returns(returnType.asClassName().parameterized())
+            .apply {
+              functionParameters.forEach { parameter ->
+                addParameter(parameter.name.asString(), parameter.type.asTypeName())
+              }
+
+              val argumentList = functionParameters.joinToString { it.name.asString() }
+              addStatement("return delegateFactory.get($argumentList)")
+            }
+            .build()
+        )
+        .apply {
+          TypeSpec.companionObjectBuilder()
+            .addFunction(
+              FunSpec.builder("create")
+                .jvmStatic()
+                .addTypeVariables(typeParameters)
+                .addParameter(delegateFactoryName, generatedFactory)
+                .returns(Provider::class.asClassName().parameterizedBy(classType))
+                .addStatement(
+                  "return %T.create(%T($delegateFactoryName))",
+                  InstanceFactory::class,
+                  implClass.parameterized()
+                )
+                .build()
+            )
+            .build()
+            .let {
+              addType(it)
+            }
```

Comment:
super nit: you have a `.let { addType(it) }` below that is all on a single line, yet this one is multiline. Make them consistent.

_(Same in `AssistedInjectGenerator`)_